habitat_sim/agent/agent.py

```diff
@@ -216,20 +232,21 @@ def set_state(
             for _, v in self._sensors.items():
                 v.set_transformation_from_spec()
 
-        for k, v in state.sensor_states.items():
-            assert k in self._sensors
-            if isinstance(v.rotation, list):
-                v.rotation = quat_from_coeffs(v.rotation)
+        if not infer_sensor_states:
+            for k, v in state.sensor_states.items():
+                assert k in self._sensors
+                if isinstance(v.rotation, list):
+                    v.rotation = quat_from_coeffs(v.rotation)
 
-            s = self._sensors[k]
+                s = self._sensors[k]
 
-            s.node.reset_transformation()
-            s.node.translate(
-                quat_rotate_vector(
-                    state.rotation.inverse(), v.position - state.position
+                s.node.reset_transformation()
+                s.node.translate(
+                    quat_rotate_vector(
+                        state.rotation.inverse(), v.position - state.position
+                    )
                 )
-            )
-            s.node.rotation = quat_to_magnum(state.rotation.inverse() * v.rotation)
+                s.node.rotation = quat_to_magnum(state.rotation.inverse() * v.rotation)
 
         if is_initial:
             self.initial_state = state
```

Comment:
Looking at the snippet in the description I first thought the `state` property assignment wasn't working correctly and thus one had to use `set_state()` instead. There's [no docs for the property](https://aihabitat.org/docs/habitat-sim/habitat_sim.agent.Agent.html#state), so maybe it could be worth adding a docstring that explains relation to `set_state()`?

Reply:
Makes sense!  Added